### PR TITLE
Resolve #1202, #1257, #1259

### LINKED ISSUE #1202: Robustness enhancement of transfer execution
To make the transfer execution more robust in case of unstable internet connection,

- send non-sensitive transfer data to chainsfr back-end before sending the transaction to the blockchain network.
- create transfer ID before broadcasting
- enhancement in chainsfr backend as well

### PR DESCRIPTION
**Pull request by timothywangdev**

### CODE REVIEW

File: src/utils.js [hunk 1]
```diff
@@ -183,7 +183,12 @@ function toCurrencyAmount (cryptoAmount: StandardTokenUnit, price: number, symbo
 }
 
 function toCryptoAmount (currencyAmount: StandardTokenUnit, price: number, symbol: ?string): string {
+  console.log(parseFloat(currencyAmount), price, parseFloat(currencyAmount) / price)
```
Comment (InVincible2016) [inline src/utils.js]: remove

File: src/apis.js [hunk 2]
```diff
@@ -55,7 +56,7 @@ async function transfer (request: {|
   sendMessage: ?string,
   cryptoType: string,
   data: string,
-  sendTxHash: TxHash,
+  sendTxHash: ?TxHash,
```
Comment (InVincible2016) [inline src/apis.js]: In what cases do we have a new transfer without a txHash?

Comment (timothywangdev) [reply src/apis.js]: backup

File: src/utils.js (hunk 1 above)
Comment (timothywangdev) [reply src/utils.js]: fixed

**Pull request merged**